src/chatty/util/history/HistoryManager.java

```diff
@@ -0,0 +1,206 @@
+     */
+    public Boolean isChannelExcluded(String channel) {
+        return settings.listContains("externalHistoryExclusion", channel);
+    }
+
+    /**
+     * Check whether the chat history feature is enabled and configured correctly.
+     *
+     * @return true if enabled and configured correctly, false otherwise
+     */
+    public boolean isEnabled() {
+        return settings.getBoolean("historyEnableService");
+    }
+
+    /**
+     * Detects input from the API with regex and transforms it into a History Message Object
+     *
+     * @param message Input from the external API
+     * @param channel A Room Object for the current channel
+     * @return A HistoryMessage Object containing all information from the historic message
+     */
+    private HistoryMessage transformStringToMessage(String message, Room channel)
+    {
+        MessageType type = MessageType.UNDEFINED;
+        HistoryMessage ret = new HistoryMessage();
+        ret.Action = false;
+        ret.Tags = MsgTags.EMPTY;
+
+        Matcher matcher = patRegexRobotty.matcher(message);
+
+        while(matcher.find()) {
+            if(matcher.group(1) != null && !matcher.group(1).isEmpty()) {
+                ret.User = new User(matcher.group(1), channel);
+            }
+            else if(matcher.group(2) != null && !matcher.group(2).isEmpty()) {
+                ret.Message = matcher.group(2);
+                type = MessageType.PRIVATMESSAGE;
+            }
+            else if(matcher.group(3) != null && !matcher.group(3).isEmpty()) {
+                ret.Tags = MsgTags.addTag(ret.Tags, "historic-timestamp", matcher.group(3));
+                ret.Timestamp = Long.parseLong(matcher.group(3));
+            }
+            else if(matcher.group(4) != null && !matcher.group(4).isEmpty()) {
+                ret.UserColor = matcher.group(4);
+            }
+        }
+
+        if(type == MessageType.PRIVATMESSAGE) {
+            ret.updateUserColor();
+            return ret;
+        }
+
+        return null;
+    }
+
+    /**
+     * Just a helper file for debugging :D
+     * @param filePath
+     * @return
+     * @throws IOException
+     */
+    private String readFileAsString(String filePath) throws IOException {
+        StringBuffer fileData = new StringBuffer();
+        BufferedReader reader = new BufferedReader(
+                new FileReader(filePath));
+        char[] buf = new char[1024];
+        int numRead=0;
+        while((numRead=reader.read(buf)) != -1) {
+            String readData = String.valueOf(buf, 0, numRead);
+            fileData.append(readData);
+        }
+        reader.close();
+        return fileData.toString();
+    }
+
+    /**
+     * Executes the actual HTTP request for historical Data
+     *
+     * @param channel Channel to start the request for
+     * @return A JSONObject with all messages requested accordingly to the parameters
+     */
+    private JSONObject executeRequest(String channel)
+    {
```

Comment:
Methods should have the opening bracket on the same line: `private JSONObject executeRequest(String channel) {`